src/internal/endpoint-runtime.ts

```diff
@@ -0,0 +1,143 @@
+import type { JsonValue } from 'type-fest'
```

Comment:
This `JsonValue` is quite restrictive - disallowing people to send some valid things such as Dates, ArrayBuffers, and more. I would probably just remove this constraint and allow the runtime to throw when people try to send invalid data. I think that replicating exact runtime requirements for this value at the type level is not worth the trouble.

Reply:
Also note that this `JsonValue` restriction is only in place for people who don't extend the `ProtocolMap`. Types provided to the `ProtocolMap` are not restricted like this, see the comment [here](https://github.com/zikaari/webext-bridge/pull/25/files#r910838194)

Reply:
I believe array buffers can't be sent over post messaging API, at least not without using transferable mechanism, which this library currently does not support